var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -199,6 +199,9 @@ def file_system_options(self):
 
     def setup_environment(self, spack_env, run_env):
         spec = self.spec
+        # mvapich2 complains during configure if F90 is set, which Intel does
+        if spec.satisfies('%intel'):
+            spack_env.unset('F90')
```

Comment:
We should always unset this, regardless of which compiler is selected. `F90FLAGS` too. Users may have Intel loaded in the background and be using `%gcc`.